### PR TITLE
[DB Schema] Create Self Check Model

### PR DESCRIPTION
**Pull request by christineNg1028**
## Notion ticket link
[API Routes (Self Check Group)](https://www.notion.so/uwblueprintexecs/API-Routes-Self-Check-Group-60d04719da5241a2ad9b6818f5cccf42)

## Implementation description
* Builds API routes for the Self Check Question and Self Check Group

## Steps to test
1. Analyze selfCheckQuestion.ts and selfCheckGroup.ts files

## What should reviewers focus on?
* Ensuring that the selfCheckQuestion.ts and selfCheckGroup.ts files follow best practice (https://mongoosejs.com/docs/schematypes.html) and work as intended

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: pages/api/models/selfCheckGroup.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+interface SelfCheckGroupInterface {
+    _id: Schema.Types.ObjectId;
+    questionIDs: [Schema.Types.ObjectId];
+}
+
+const SelfCheckGroupSchema = new Schema<SelfCheckGroupInterface>(
+    {
+        questionIDs: {
+            type: [{ type: Schema.Types.ObjectId, ref: "Questions" }],
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckGroup.ts:11]: for ref do you mean `SelfCheckQuestion`?

File: pages/api/models/selfCheckGroup.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            default: [],
+        },
+    },
+    {
+        timestamps: {
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckGroup.ts:16]: lets just use `{timestamps: true}`

File: pages/api/models/selfCheckQuestion.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            type: QuestionType,
+            required: true,
+        },
+        question: {
+            type: String,
+            required: true,
+            trim: true,
+        },
+        options: {
+            type: [{ type: Schema.Types.Mixed, ref: "Options" }],
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckQuestion.ts:28]: Is `Options` a model?

File: pages/api/models/selfCheckQuestion.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        },
+        questionNumber: {
+            type: Number,
+            required: true,
+        },
+    },
+    {
+        timestamps: {
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckQuestion.ts:36]: {timestamps: true}

File: pages/api/models/selfCheckGroup.ts (hunk 2 above)
Comment (nimwijetunga) [reply pages/api/models/selfCheckGroup.ts:16]: This just gets the createdAt and updatedAt fields (which is prob the only ones we need)

File: pages/api/models/selfCheckGroup.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+            currentTime: Date.now,
+            updatedAt: "dateUpdated",
+            createdAt: "dateCreated",
+        },
+    },
```
Comment (daniel-sm-yu) [inline pages/api/models/selfCheckGroup.ts:21]: `{ timestamps: true }` should be sufficient (https://masteringjs.io/tutorials/mongoose/timestamps)

**Review by nimwijetunga — CHANGES_REQUESTED** [review]
Can you also test these models through the dbTest endpoint (i.e create an instance of SelfCheckQuestion and SelfCheckGroup) to make sure that they work?

File: pages/api/models/selfCheckGroup.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+interface SelfCheckGroupInterface {
+    _id: Types.ObjectId;
+    questionIDs: [Types.ObjectId];
+}
+
+const SelfCheckGroupSchema = new Schema<SelfCheckGroupInterface>(
+    {
+        questionIDs: {
+            type: [{ type: Types.ObjectId, ref: "SelfCheckQuestion" }],
+            default: [],
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckGroup.ts:12]: I belive arrays are implicity [] by default so you can remove this

File: pages/api/models/selfCheckQuestion.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+const SelfCheckQuestionSchema = new Schema<SelfCheckQuestionInterface>(
+    {
+        type: {
+            type: QuestionType,
+            required: true,
+        },
+        question: {
+            type: String,
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckQuestion.ts:23]:
```suggestion
            type: string,
```

File: pages/api/models/selfCheckQuestion.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+            required: true,
+            trim: true,
+        },
+        options: {
+            type: [Schema.Types.Mixed],
+            default: [],
+        },
+        questionNumber: {
+            type: Number,
```
Comment (nimwijetunga) [inline pages/api/models/selfCheckQuestion.ts:32]:
```suggestion
            type: number,
```

**Review by nimwijetunga — APPROVED** [review]
lgtm!

File: pages/api/models/selfCheckQuestion.ts (hunk 8 above)
Comment (tawnyzhao) [reply pages/api/models/selfCheckQuestion.ts:32]: Number is correct actually you pass the constructor to the Schema https://mongoosejs.com/docs/schematypes.html#numbers

File: pages/api/models/selfCheckQuestion.ts (hunk 7 above)
Comment (tawnyzhao) [reply pages/api/models/selfCheckQuestion.ts:23]: String is correct https://mongoosejs.com/docs/schematypes.html#strings

**Review by daniel-sm-yu — APPROVED** [review]

File: pages/api/models/selfCheckQuestion.ts [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    type: QuestionType;
+    question: string;
+    options: [Schema.Types.Mixed];
+    questionNumber: number;
+}
+
+const SelfCheckQuestionSchema = new Schema<SelfCheckQuestionInterface>(
+    {
+        type: {
+            type: QuestionType,
```
Comment (tawnyzhao) [inline pages/api/models/selfCheckQuestion.ts:19]: For enum types, you want String as the type with the enum field.
```suggestion
            type: String,
            enum: Object.values(QuestionType)
```
See: https://mongoosejs.com/docs/schematypes.html#schematype-options  
https://mongoosejs.com/docs/validation.html#custom-validators

**Review by tawnyzhao — CHANGES_REQUESTED** [review]

**Pull request merged**